File: Tests/SwiftKotlinFrameworkTests/Tests/plugins/XCTTestToJUnitTokenTransformPlugin.swift

import XCTest
@testable import MySportsSDK

class AvailabilityServiceTests: XCTestCase {
    var service: AvailabilityService!
    var mockHttpClient: MockHttpClient!

    override func setUp() {
        super.setUp()
        mockHttpClient = MockHttpClient()
        service = AvailabilityService(httpClient: mockHttpClient)
    }

    override func tearDown() {
        mockHttpClient.clear()
        super.tearDown()
    }

    func testAssertEqual() {
        XCTAssertEqual(1, 1)
        XCTAssertEqual(1, 1, "Not equal")
    }

    func testAssertTrue() {
        XCTAssertTrue(true)
        XCTAssertTrue(true, "Not true")
    }

    func testAssertFalse() {
        XCTAssertFalse(false)
        XCTAssertFalse(false, "Should not be true")
    }

    func testAssertNil() {
        XCTAssertNil(nil)
        XCTAssertNil(nil, "Should be nil")
    }

    func testAssertNotNil() {
        XCTAssertNotNil("")
        XCTAssertNotNil("", "Should not be nil")
    }

    func testMultipleStatements() {
        mockHttpClient.fileResponse = "empty"
        let result = service.search()
        let params = mockHttpClient.parameters

        XCTAssertTrue(mockHttpClient.method == "GET", "Expected get request")
        XCTAssertEqual(mockHttpClient.endpoint, "/api/v2/availability/search")
        XCTAssertNotNil(params, "Expected parameters")
        XCTAssertNil(promise.error, "Promise should not have error")
        XCTAssertNotNil(result)
    }

    func noTestMethod() {
        XCTAssertTrue(true)
    }

}

class TestWithInheritance: XCTestCase, SomeProtocol {
    func testMethodNotUnderXCTest() {
        XCTAssertTrue(true)
    }
}

class ClassWithNoTest: Other {

    override func setUp() {
        super.setUp()
        mockHttpClient = MockHttpClient()
        service = AvailabilityService(httpClient: mockHttpClient)
    }

    override func tearDown() {
        mockHttpClient.clear()
        super.tearDown()
    }

    func testMethodNotUnderXCTest() {
        XCTAssertTrue(true)
    }
}

File: Tests/SwiftKotlinFrameworkTests/Tests/plugins/XCTTestToJUnitTokenTransformPlugin.kt
import org.junit.*
import org.junit.Assert.*
class AvailabilityServiceTests {
    lateinit var service: AvailabilityService
    lateinit var mockHttpClient: MockHttpClient
    @Before
    fun setUp() {
        mockHttpClient = MockHttpClient()
        service = AvailabilityService(httpClient = mockHttpClient)
    }
    @After
    fun tearDown() {
        mockHttpClient.clear()
    }
    @Test
    fun testAssertEqual() {
        XCTAssertEqual(1, 1)
        XCTAssertEqual(1, 1, "Not equal")
    }
    @Test
    fun testAssertTrue() {
        XCTAssertTrue(true)
        XCTAssertTrue(true, "Not true")
    }
    @Test
    fun testAssertFalse() {
        XCTAssertFalse(false)
        XCTAssertFalse(false, "Should not be true")
    }
    @Test
    fun testAssertNil() {
        XCTAssertNil(null)
        XCTAssertNil(null, "Should be nil")
    }
    @Test
    fun testAssertNotNil() {
        XCTAssertNotNil("")
        XCTAssertNotNil("", "Should not be nil")
    }
    @Test
    fun testMultipleStatements() {
        mockHttpClient.fileResponse = "empty"
        val result = service.search()
        val params = mockHttpClient.parameters
        XCTAssertTrue(mockHttpClient.method == "GET", "Expected get request")
        XCTAssertEqual(mockHttpClient.endpoint, "/api/v2/availability/search")
        XCTAssertNotNil(params, "Expected parameters")
        XCTAssertNil(promise.error, "Promise should not have error")
        XCTAssertNotNil(result)
    }
    fun noTestMethod() {
        XCTAssertTrue(true)
    }
}
class TestWithInheritance: SomeProtocol {
    @Test
    fun testMethodNotUnderXCTest() {
        XCTAssertTrue(true)
    }
}
class ClassWithNoTest: Other {
    override fun setUp() {
        super.setUp()
        mockHttpClient = MockHttpClient()
        service = AvailabilityService(httpClient = mockHttpClient)
    }
    override fun tearDown() {
        mockHttpClient.clear()
        super.tearDown()
    }
    fun testMethodNotUnderXCTest() {
        XCTAssertTrue(true)
    }
}
